src/Squirrel.Tool/Program.cs

```diff
@@ -18,52 +23,76 @@ class Program
 
         private static ConsoleLogger _logger;
 
-        static int Main(string[] inargs)
+        private static Option<string> CsqVersion { get; } 
+            = new Option<string>("--csq-version");
+        private static Option<FileSystemInfo> CsqSolutionPath { get; } 
+            = new Option<FileSystemInfo>(new[] { "--csq-sln", "--csq-solution" }).ExistingOnly();
+        private static Option<bool> Verbose { get; }
+            = new Option<bool>("--verbose");
+
+        static Task<int> Main(string[] inargs)
         {
             _logger = ConsoleLogger.RegisterLogger();
-            try {
-                return MainInner(inargs);
-            } catch (Exception ex) {
-                Console.WriteLine("csq error: " + ex.Message);
-                return -1;
+
+            RootCommand rootCommand = new RootCommand() {
+                CsqVersion,
+                CsqSolutionPath,
+                Verbose
+            };
+            rootCommand.TreatUnmatchedTokensAsErrors = false;
+
+            rootCommand.SetHandler(MainInner);
+
+            ParseResult parseResult = rootCommand.Parse(inargs);
+
+            CommandLineBuilder builder = new CommandLineBuilder(rootCommand);
+
+            if (parseResult.Directives.Contains("local")) {
```

Comment:
A [directive](https://learn.microsoft.com/dotnet/standard/commandline/syntax#directives) is a metadata level item. 
This allows for being able to interact with this tool's help directly by running `csq [local] --help` to view the help of the `csq` tool directly, while running `csq --help` will display the help of the found squirrel utility.

Reply:
This is great, we'll need to document it. I have started re-working all the documentation in the `cs/newdocs` branch if you're keen to have a look - I'll add a note about this there.